tests/index/crlf.c

```diff
@@ -43,84 +43,61 @@ struct compare_data
 
 static int add_and_check_file(void *payload, git_buf *actual_path)
 {
-	git_buf expected_path_oid = GIT_BUF_INIT;
+	git_buf expected_path = GIT_BUF_INIT;
 	git_buf expected_path_fail = GIT_BUF_INIT;
 	git_buf expected_contents = GIT_BUF_INIT;
 	struct compare_data *cd = payload;
-	bool failed = true;
-	int cmp_git, cmp_gitattributes;
 	char *basename;
-	int add_bypath_ret;
 	const git_index_entry *entry;
-	git_oid oid;
-	git_error_state error = { 0 };
+	git_blob *blob;
 
 	basename = git_path_basename(actual_path->ptr);
-	cmp_git = strcmp(basename, ".git");
-	cmp_gitattributes = strcmp(basename, ".gitattributes");
-
-	if (cmp_git == 0 || cmp_gitattributes == 0) {
-		failed = false;
-		goto done;
-	}
 
-	giterr_clear();
-	add_bypath_ret = git_index_add_bypath(g_index, basename);
-	giterr_state_capture(&error, add_bypath_ret);
+	if (!strcmp(basename, ".git") || !strcmp(basename, ".gitattributes"))
+		return 0;
 
-	entry = git_index_get_bypath(g_index, basename, 0);
-	if (!add_bypath_ret && !entry)
-		goto done;
+	cl_git_pass(git_buf_joinpath(&expected_path, cd->dirname, basename));
 
-	cl_git_pass(git_buf_joinpath(&expected_path_oid, cd->dirname, basename));
-	cl_git_pass(git_buf_joinpath(&expected_path_fail, cd->dirname, basename));
-	git_buf_puts(&expected_path_oid, ".obj");
-	git_buf_puts(&expected_path_fail, ".fail");
+	cl_git_pass(git_buf_puts(&expected_path_fail, expected_path.ptr));
+	cl_git_pass(git_buf_puts(&expected_path_fail, ".fail"));
 
-	if (git_path_isfile(expected_path_oid.ptr)) {
-		if (add_bypath_ret)
-			goto done;
+	if (git_path_isfile(expected_path.ptr)) {
+		cl_git_pass(git_index_add_bypath(g_index, basename));
 
-		if (git_futils_readbuffer(&expected_contents, expected_path_oid.ptr) < 0)
-			goto done;
+		cl_assert(entry = git_index_get_bypath(g_index, basename, 0));
+		cl_git_pass(git_blob_lookup(&blob, g_repo, &entry->id));
 
-		if (git_oid_fromstr(&oid, expected_contents.ptr))
-			goto done;
-		if (!git_oid_equal(&oid, &entry->id))
-			goto done;
-	}
-
-	if (git_path_isfile(expected_path_fail.ptr)) {
-		if (!add_bypath_ret)
-			goto done;
-
-		if (git_futils_readbuffer(&expected_contents, expected_path_fail.ptr) < 0)
-			goto done;
+		cl_git_pass(git_futils_readbuffer(&expected_contents, expected_path.ptr));
+		cl_assert_equal_s(expected_contents, git_blob_rawcontent(blob));
 
+		git_blob_free(blob);
+	} else if (git_path_isfile(expected_path_fail.ptr)) {
+		cl_git_pass(git_futils_readbuffer(&expected_contents, expected_path_fail.ptr));
 		git_buf_rtrim(&expected_contents);
 
-		if (error.error_msg.klass != GITERR_FILTER || strstr(error.error_msg.message, expected_contents.ptr) == NULL)
-			goto done;
+		cl_git_fail(git_index_add_bypath(g_index, basename));
+		cl_assert_equal_i(GITERR_FILTER, giterr_last()->klass);
+		cl_assert_equal_s(expected_contents.ptr, giterr_last()->message);
+	} else {
+		cl_fail("unexpected index failure");
```

Comment:
Ugh. Okay, so you do already fix up a lot of these. Maybe want to squash those two commits together?

Reply:
Yeah, I did that so that @csware's commits kept him as the author to make sure that he got credit in the log.

Reply:
Thanks ;)